lstchain/scripts/lstchain_dl1ab_pedestal_cleaning.py

```diff
@@ -0,0 +1,186 @@
+    pic_thresh = config['tailcuts_clean_with_pedestal_threshold']['picture_thresh']
+    bound_thresh = config['tailcuts_clean_with_pedestal_threshold']['boundary_thresh']
+    sigma = config['tailcuts_clean_with_pedestal_threshold']['sigma']
+    keep_isolated_pixels = config['tailcuts_clean_with_pedestal_threshold']['keep_isolated_pixels']
+    min_number_picture_neighbors = config['tailcuts_clean_with_pedestal_threshold']['min_number_picture_neighbors']
+
+    pedestal_threshhold = get_threshold_from_dl1_file(args.input_file, sigma)
+    picture_thresh = np.clip(pedestal_threshhold, pic_thresh, None)
+
+    foclen = OpticsDescription.from_name('LST').equivalent_focal_length
+    cam_table = Table.read(args.input_file, path="instrument/telescope/camera/LSTCam")
+    camera_geom = CameraGeometry.from_table(cam_table)
+
+    dl1_container = DL1ParametersContainer()
+    parameters_to_update = list(HillasParametersContainer().keys())
+    parameters_to_update.extend([
+        'concentration_cog',
+        'concentration_core',
+        'concentration_pixel',
+        'leakage_intensity_width_1',
+        'leakage_intensity_width_2',
+        'leakage_pixels_width_1',
+        'leakage_pixels_width_2',
+        'n_islands',
+        'intercept',
+        'time_gradient',
+        'n_pixels',
+        'wl',
+        'r',
+        'log_intensity',
+        'x',
+        'y'
+    ])
+
+    nodes_keys = get_dataset_keys(args.input_file)
+    if args.noimage:
+        nodes_keys.remove(dl1_images_lstcam_key)
+
+    auto_merge_h5files([args.input_file], args.output_file, nodes_keys=nodes_keys)
+
+    with tables.open_file(args.input_file, mode='r') as input:
+        image_table = input.root[dl1_images_lstcam_key]
+        with tables.open_file(args.output_file, mode='a') as output:
+
+            params = output.root[dl1_params_lstcam_key].read()
+
+            for ii, row in enumerate(image_table):
+
+                dl1_container.reset()
+
+                image = row['image']
+                peak_time = row['peak_time']
+
+                signal_pixels = tailcuts_clean(
+                                    camera_geom,
+                                    image,
+                                    picture_thresh,
+                                    bound_thresh,
+                                    keep_isolated_pixels,
+                                    min_number_picture_neighbors
+                                    )
+                n_pixels = np.count_nonzero(signal_pixels)
+                if n_pixels > 0:
+                    num_islands, island_labels = number_of_islands(camera_geom, signal_pixels)
+                    n_pixels_on_island = np.bincount(island_labels.astype(np.int))
+                    n_pixels_on_island[0] = 0  # first island is no-island and should not be considered
+                    max_island_label = np.argmax(n_pixels_on_island)
+                    signal_pixels[island_labels != max_island_label] = False
```

Comment:
This should be configurable with a command-line option or through config file. With this star-aware cleaning we probably will do away with the "main island" parameters, and use all cleaning-surviving pixels instead.